### PR TITLE
Sn-969 Revise ROS Node to add correct odom messages and INS type messages

### PR DESCRIPTION
**Pull request by andrewpriddis**

### CODE REVIEW

**Comment by waltjohnson** [comment]
@siddirp This looks great.  Before we merge it into development, let's make sure we've tested it in Luna to ensure the outputs look like we expect.

**Review by waltjohnson — APPROVED** [review]
@siddirp I approve.  However, let's not merge this until it is thoroughly tested according to the checklist in the jira card.

@aabogdanov Alex needs to also double check the transformations.

File: src/inertial_sense_ros.cpp [hunk 1]
```diff
@@ -5,857 +5,1227 @@
...
+    {
+        odom_ins_ecef_.pub = nh_.advertise<nav_msgs::Odometry>("odom_ins_ecef", 1);
+        SET_CALLBACK(DID_INS_4, ins_4_t, INS4_callback, 1);                                                   // Need quaternion and ecef
+        if (covariance_enabled_)
+            SET_CALLBACK(DID_ROS_COVARIANCE_POSE_TWIST, ros_covariance_pose_twist_t, INS_covariance_callback, 1); // Need Covariance data
+        SET_CALLBACK(DID_PREINTEGRATED_IMU, preintegrated_imu_t, preint_IMU_callback, 1);                                              // Need angular rate data from IMU
+        IMU_.enabled = true;
+        // Create Identity Matrix
+        //
+        for (int row = 0; row < 6; row++)
```
Comment (aabogdanov) [inline src/inertial_sense_ros.cpp:119]: SDK ISMatrix.c has a function that creates an identity matrix, **void eye_MatN( f_t *A, i_t n )**. We could probably use it here.

File: src/inertial_sense_ros.cpp [hunk 2]
```diff
@@ -5,857 +5,1227 @@
...
+
+            ned_odom_msg.twist.twist.linear.x = result[0];
+            ned_odom_msg.twist.twist.linear.y = result[1];
+            ned_odom_msg.twist.twist.linear.z = result[2];
+
+            // Angular Velocity
+
+            // Transform from body frame to NED
+            ixVector3 angVelImu = {(f_t)imu_msg.angular_velocity.x, (f_t)imu_msg.angular_velocity.y, (f_t)imu_msg.angular_velocity.z};
+            quatConjRot(result, qn2b, angVelImu);
```
Comment (aabogdanov) [inline src/inertial_sense_ros.cpp]: I think it needs to be quatRot() here. 
Here is its description from ISPose.c
/* Computationally simple means to apply quaternion rotation to a vector.
 * Requires quaternion be normalized first.
 * If quaternion describes current attitude, then **rotation is body -> inertial frame**.
 * Equivalent to a DCM.T * vector multiply.
 */
void quatRot(ixVector3 result, const ixQuat q, const ixVector3 v)

File: src/inertial_sense_ros.cpp [hunk 3]
```diff
@@ -5,857 +5,1227 @@
...
+
+            enu_odom_msg.twist.twist.linear.x = result[1];
+            enu_odom_msg.twist.twist.linear.y = result[0];
+            enu_odom_msg.twist.twist.linear.z = -result[2];
+
+            // Angular Velocity
+
+            // Transform from body frame to ENU
+            ixVector3 angVelImu = {(f_t)imu_msg.angular_velocity.x, (f_t)imu_msg.angular_velocity.y, (f_t)imu_msg.angular_velocity.z};
+            quatConjRot(result, qenu2b, angVelImu);
```
Comment (aabogdanov) [inline src/inertial_sense_ros.cpp]: I think it should be quatRot() here (rotation from body to reference frame)

**Review by aabogdanov — CHANGES_REQUESTED** [review]

File: src/inertial_sense_ros.cpp (hunk 3 above)
Comment (andrewpriddis) [reply src/inertial_sense_ros.cpp]: OK

File: src/inertial_sense_ros.cpp (hunk 1 above)
Comment (andrewpriddis) [reply src/inertial_sense_ros.cpp:119]: OK

Comment (andrewpriddis) [reply src/inertial_sense_ros.cpp:119]: DIdn't work for doubles

File: src/inertial_sense_ros.cpp (hunk 2 above)
Comment (andrewpriddis) [reply src/inertial_sense_ros.cpp]: OK

**Pull request merged**